lib/devise/models/authenticatable.rb

```diff
@@ -253,6 +253,7 @@ def find_or_initialize_with_error_by(attribute, value, error=:invalid) #:nodoc:
 
         # Find or initialize a record with group of attributes based on a list of required attributes.
         def find_or_initialize_with_errors(required_attributes, attributes, error=:invalid) #:nodoc:
+          attributes = attributes.to_unsafe_h.with_indifferent_access if attributes.respond_to? :to_unsafe_h
```

Comment:
I'm figuring it out on how to stop passing an `ActionController::Parameters` to this method, but I'll work on it on a new PR later so we can remove this.